spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/cql/RefCountedCache.scala

```diff
@@ -53,6 +53,18 @@ final class RefCountedCache[K, V](create: K => V,
         else
           acquire(key)
       case None =>
+        syncAcquire(key)
+    }
+  }
+
+  private[this] def syncAcquire(key: K): V = synchronized {
+    cache.get(key) match {
+      case Some(value) =>
+        if (refCounter.acquireIfNonZero(value) > 0)
+          value
+        else
+          acquire(key)
+      case None =>
```

Comment:
Hm, I can not comment the code that out of diff,so:
76 line: The code should not be reachable in sync block as I understand. SO let replace that block with the
ConcurentModificationException.